clients/wallet/src/cache.rs

```diff
@@ -0,0 +1,344 @@
+use primitives::TransactionEnvelopeExt;
+use std::{
+	fs::{create_dir_all, read_dir, remove_file, File, OpenOptions},
+	io::{Read, Write},
+	path::Path,
+};
+use substrate_stellar_sdk::{types::SequenceNumber, TransactionEnvelope, XdrCodec};
+
+/// a helpful macro to unwrap an `Ok` or return immediately.
+macro_rules! unwrap_or_return {
+	// expression, return value, extra log
+	($res:expr, $ret:expr, $log:expr) => {
+		match $res {
+			Ok(result) => result,
+			Err(e) => {
+				tracing::warn!("{:?}: {:?}", $log, e);
+				return $ret
+			},
+		}
+	};
+	($res:expr, $ret:expr) => {
+		match $res {
+			Ok(result) => result,
+			Err(e) => {
+				tracing::warn!("{:?}", e);
+				return $ret
+			},
+		}
+	};
+	($res:expr) => {
+		match $res {
+			Ok(result) => result,
+			Err(e) => {
+				tracing::warn!("{:?}", e);
+				return
+			},
+		}
+	};
+}
+
+/// Contains the path where the transaction envelope will be saved.
+#[derive(Debug, Clone)]
+pub struct TxEnvelopeStorage {
+	path: String,
+	inner_path: String,
+}
+
+impl TxEnvelopeStorage {
+	/// returns the full directory path
+	fn full_path(&self) -> String {
+		format!("{}/{}", self.path, self.inner_path)
+	}
+}
+
+impl TxEnvelopeStorage {
+	pub fn new(path: String, public_key: &str, is_public_network: bool) -> Self {
+		let inner_path = format!("{public_key}_{is_public_network}");
+		let full_path = format!("{path}/{public_key}_{is_public_network}");
+
+		if let Err(e) = create_dir_all(&full_path) {
+			tracing::warn!("Failed to create directory of {full_path}: {:?}", e);
+		}
+		tracing::info!("path for caching stellar transactions: {full_path}");
+
+		TxEnvelopeStorage { path, inner_path }
+	}
+
+	/// Returns true if transaction was saved to a local path.
+	/// The filename will be the transaction's sequence number.
+	pub fn save_to_local(
+		&self,
+		sequence: SequenceNumber,
+		transaction: TransactionEnvelope,
+	) -> bool {
+		let full_file_path = format!("{}/{}", self.full_path(), sequence);
+
+		let path = Path::new(&full_file_path);
+		if path.exists() {
+			return false
+		}
+
+		let mut file = unwrap_or_return!(
+			OpenOptions::new().write(true).create(true).open(path),
+			false,
+			format!("Failed to create file for transaction with sequence {}", sequence)
+		);
+
+		unwrap_or_return!(
+			// writing the transaction to file in Vec<u8> format
+			write!(file, "{:?}", transaction.to_xdr()),
+			false,
+			format!("Failed to write transaction with sequence {}", sequence)
+		);
+
+		true
+	}
+
+	pub fn remove_transaction(&self, sequence: SequenceNumber) -> bool {
+		let full_file_path = format!("{}/{sequence}", self.full_path());
+		remove_file(full_file_path).is_ok()
+	}
+
+	#[doc(hidden)]
+	#[cfg(any(test, feature = "testing-utils"))]
+	/// Removes all transactions, including the directory
+	pub fn remove_all(&self) -> bool {
+		std::fs::remove_dir_all(self.full_path()).is_ok()
+	}
+
+	#[allow(dead_code)]
+	#[cfg(any(test, feature = "testing-utils"))]
```

Comment:
It's useful for testing, but highly unlikely that we want to access one single envelope for other purpose, aside from resubmitting it. And when we resubmit, it's all or nothing. 

It's nice to have tbh, but it's also a dead code since no actual logic is calling this method (again, aside from testing).

Reply:
Okay, fine to keep it like that then